internal/dataplane/parser/ingressrules.go

```diff
@@ -283,13 +284,17 @@ func getK8sServicesForBackends(
 	return k8sServices, seenAnnotationsForK8sServices
 }
 
-func servicesAllUseTheSameKongAnnotations(
+// collectInconsistentAnnotations takes a list of services and annotation+value pairs and confirms that all services
+// have those annotations with those values. If any service does not have one of the annotation+value pairs, push
+// a resource failure to the provided collector for all services indicating the problem annotation.
+func collectInconsistentAnnotations(
 	services []*corev1.Service,
 	annotations map[string]string,
 	failuresCollector *failures.ResourceFailuresCollector,
 	kongServiceName string,
 ) bool {
 	match := true
+	badAnnotations := sets.Set[string]{}
```

Comment:
Just a note that there's also `sets.NewString()` where you don't need to provide the type param. It's basically calls `sets.Set[string]{}` underneath but I thought it will be worth mentioning.